### PR TITLE
Adding in new API endpoints

### PR DESCRIPTION
**Pull request by morganick**
- need to have flattened API endpoint go into production on app side

### CODE REVIEW

**Comment by morganick** [comment]
@ryansch @gjamerson please review

**Comment by gjamerson** [comment]
:+1:

**Comment by morganick** [comment]
connected to outstand/app#1564

File: endpoints/saved_messages.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+* `GET /saved_messages.json` returns a paginated list of saved messages for the user
+
+Returns a `200 OK` with an array of contacts. (See [Saved Message](#saved-message) for JSON representation.) If you request a page of saved messages that does not exist, a `404 Not Found` will be returned. (See [Pagination](https://github.com/outstand/api-docs#pagination))
+
+### Optional Query Params
+
+* `template` supports the slug of the template: `branded-email`, `email-greeting`, `newsletter-single-column`, `newsletter-two-column`, `greeting-card`, `mini-greeting-card`, `postcard`, `mini-postcard` (`email-greeting` is default.)
+* `q` search query
+* `category`
+* `filter_by` supports one of the following: `all`, `mine`, `buy`. (`all` is default.) `mine` filters saved messages to only saved messages that the user created, belong to the user's white label group, or the user has purchased.
```
Comment (ryansch) [inline endpoints/saved_messages.md]: `buy` needs documentation

**Comment by ryansch** [comment]
Comments left

**Pull request merged**